tests/object/lookup.c

```diff
@@ -48,6 +48,19 @@ void test_object_lookup__lookup_wrong_type_by_abbreviated_id_returns_enotfound(v
 		GIT_ENOTFOUND, git_object_lookup_prefix(&object, g_repo, &oid, strlen(commit), GIT_OBJ_TAG));
 }
 
+void test_object_lookup__lookup_with_abbreviated_oid_sets_trailing_bits_to_zero(void)
+{
+	const char *commit = "e90810b8df3e80c413d903f631643c716887138d";
+	git_oid oid;
+	git_object *object;
+
+	cl_git_pass(git_oid_fromstrn(&oid, commit, strlen(commit)));
+	cl_git_pass(git_object_lookup_prefix(&object, g_repo, &oid, GIT_OID_HEXSZ - 1, GIT_OBJ_COMMIT));
+	git_object_free(object);
+
+	cl_assert_equal_i(0, oid.id[GIT_OID_HEXSZ - 1]);
```

Comment:
The id is purely an input parameter. Nothing we do inside `git_object_lookup_prefix()` would modify what this value contains. Even if we did, the size is in nibbles so this byte wouldn't be fully zeroed out, just the lower half. It is rather surprising that this succeeds in most builds.